### PR TITLE
Share 2D OD example with stratification by bbox area

### PR DESCRIPTION
**Pull request by munkyshi**
### Linked issue(s):
Fixes: KOL-2726

### What change does this PR introduce and why?
Provide an example within 2D OD with usage of the `ignored_bboxes` field

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/workflow/annotation.py [hunk 1]
```diff
@@ -77,6 +77,11 @@ class BoundingBox(Annotation):
     def _data_type() -> _AnnotationType:
         return _AnnotationType.BOUNDING_BOX
 
+    def area(self) -> float:
+        width = self.bottom_right[0] - self.top_left[0]
+        height = self.bottom_right[1] - self.top_left[1]
+        return width * height
```
Comment (munkyshi) [inline kolena/workflow/annotation.py]: Would love a sanity check on if there's a specific reason we didn't want this attached to `kolena.workflow.BoundingBox`?

File: examples/object_detection_2d/object_detection_2d/seed_test_suite.py [hunk 2]
```diff
@@ -134,14 +134,53 @@ def seed_test_suite_by_brightness(test_suite_name: str, complete_test_case: Test
...
 def main(args: Namespace) -> None:
     kolena.initialize(os.environ["KOLENA_TOKEN"], verbose=True)
 
     complete_test_case = create_complete_transportation_case(args)
 
     # organize test suite names with its generator
     test_suites: List[Tuple[str, Callable[[str, TestCase], TestSuite]]] = [
-        (f"{DATASET} :: transportation brightness [Object Detection]", seed_test_suite_by_brightness),
+        (f"{DATASET} :: transportation by brightness [Object Detection]", seed_test_suite_by_brightness),
+        (f"{DATASET} :: transportation by bounding box size [Object Detection]", seed_test_suite_by_bounding_box_size),
```
Comment (munkyshi) [inline examples/object_detection_2d/object_detection_2d/seed_test_suite.py:190]: Running through this on our trunk stack right now -- hopefully a useful example coming up soon.

File: examples/object_detection_2d/object_detection_2d/seed_test_suite.py [hunk 3]
```diff
@@ -134,14 +134,53 @@ def seed_test_suite_by_brightness(test_suite_name: str, complete_test_case: Test
...
+            bboxes.append(bbox) if filter_fn(bbox.area()) else ignored_bboxes.append(bbox)
+        return GroundTruth(bboxes=bboxes, ignored_bboxes=ignored_bboxes)
+
+    # create each test case by stratification
+    test_cases: List[TestCase] = []
+    for name, fn in stratification_logic_map.items():
+        samples_with_filtered_bboxes = []
+        for ts, gt in complete_test_case.iter_test_samples():
+            filtered_ground_truth = filter_gt_bboxes(gt, fn)
+            if filtered_ground_truth.n_bboxes > 0:
```
Comment (ly-kolena) [inline examples/object_detection_2d/object_detection_2d/seed_test_suite.py:157]: does `filter_gt_bboxes` set `n_bboxes`?

Comment (gordonhart) [reply examples/object_detection_2d/object_detection_2d/seed_test_suite.py:157]: `n_bboxes` is [automatically derived](https://github.com/kolenaIO/kolena/blob/trunk/kolena/_experimental/object_detection/workflow.py#L49)

**Review by gordonhart — APPROVED** [review]

File: examples/object_detection_2d/object_detection_2d/seed_test_suite.py (hunk 3 above)
Comment (ly-kolena) [reply examples/object_detection_2d/object_detection_2d/seed_test_suite.py:157]: nice

**Pull request merged**